corehq/project_limits/models.py

```diff
@@ -20,3 +22,17 @@ def delete(self, *args, **kwargs):
     def _clear_caches(self):
         from corehq.project_limits.rate_limiter import get_dynamic_rate_definition
         get_dynamic_rate_definition.clear(self.key, {})
+
+
+@architect.install('partition', type='range', subtype='date', constraint='week', column='timestamp')
+class RateLimitedTwoFactorLog(models.Model):
+    date = models.DateTimeField(auto_now_add=True)
+    username = models.CharField(max_length=255, null=False, db_index=True)
```

Comment:
@dannyroberts do you have any thoughts on which columns should be indexed? I figured most queries will be for specific IPs, numbers, or users.

Reply:
We probably want to index by date so that it's cheap to order by date or query just the last X days